consensus/src/main/kotlin/maru/consensus/qbft/QbftManager.kt

```diff
@@ -0,0 +1,165 @@
+private const val FUTURE_MESSAGES_LIMIT = 1000L // TODO: Make this configurable
+
+class QbftManager(
+  private val beaconChain: BeaconChain,
+  private val maruConfig: MaruConfig,
+  private val metricsSystem: MetricsSystem,
+  private val executionLayerManager: ExecutionLayerManager,
+  private val forksSchedule: maru.consensus.ForksSchedule,
+) {
+  fun start() {
+    val validatorKey = maruConfig.validator?.validatorKey ?: throw IllegalArgumentException("Validator key not found")
+    val signatureAlgorithm = SignatureAlgorithmFactory.getInstance()
+    val privateKey = signatureAlgorithm.createPrivateKey(Bytes32.wrap(validatorKey))
+    val keyPair = signatureAlgorithm.createKeyPair(privateKey)
+    val securityModule = KeyPairSecurityModule(keyPair)
+    val nodeKey = NodeKey(securityModule)
+
+    val blockChain = QbftBlockchainAdapter(beaconChain)
+
+    val localAddress = Util.publicKeyToAddress(nodeKey.publicKey)
+    val staticValidatorProvider = StaticValidatorProvider(setOf(Validator(localAddress.toArrayUnsafe())))
+    val validatorProvider = QbftValidatorProviderAdapter(staticValidatorProvider)
+    val proposerSelector = ProposerSelectorAdapter(ProposerSelectorImpl(beaconChain, staticValidatorProvider))
+    val validatorMulticaster = NoopValidatorMulticaster()
+    val qbftBlockCreatorFactory =
+      QbftBlockCreatorFactoryImpl(executionLayerManager, proposerSelector, staticValidatorProvider, beaconChain)
+
+    // TODO create besu forksSchedule from maru forksSchedule
+    val forksSchedule = ForksSchedule<BftConfigOptions>(emptyList())
+
+    val clock = Clock.systemUTC()
+    val bftExecutors = BftExecutors.create(metricsSystem, BftExecutors.ConsensusType.QBFT)
+    val bftEventQueue = BftEventQueue(MESSAGE_QUEUE_LIMIT)
+    val roundTimer = RoundTimer(bftEventQueue, ROUND_EXPIRY, bftExecutors)
+    val blockTimer = BlockTimer(bftEventQueue, forksSchedule, bftExecutors, clock)
+    val finalState =
+      QbftFinalStateAdapter(
+        localAddress,
+        nodeKey,
+        validatorProvider,
+        proposerSelector,
+        validatorMulticaster,
+        roundTimer,
+        blockTimer,
+        qbftBlockCreatorFactory,
+        clock,
+      )
+
+    // TODO connect this to the maru NewBlockHandler
+    val minedBlockObservers = Subscribers.create<QbftMinedBlockObserver>()
+
+    val blockCodec = QbftBlockCodecAdapter()
+    val blockInterface = QbftBlockInterfaceAdapter()
+    val protocolSchedule = QbftProtocolScheduleAdapter()
+    val messageValidatorFactory =
+      MessageValidatorFactory(proposerSelector, protocolSchedule, validatorProvider, blockInterface)
+    val messageFactory = MessageFactory(nodeKey, blockCodec)
+    val qbftRoundFactory =
+      QbftRoundFactory(
+        finalState,
+        blockInterface,
+        protocolSchedule,
+        minedBlockObservers,
+        messageValidatorFactory,
+        messageFactory,
+      )
+
+    val transitionLogger = QbftValidatorModeTransitionLoggerAdapter()
```

Comment:
What kind of transition is it? From a non-validator to a validator and vise versa?

Reply:
It's a transition to a new block. I don't think we need this interface; we should be able to remove it and just add a subscriber to the new block event in Besu. If I can do that, I'll remove this as I don't like this `QbftValidatorModeTransitionLogger`.